examples/structured_data/movielens_recommendations_transformers.py

```diff
@@ -292,198 +353,206 @@ def process(features):
+        def embedding_helper(input):
+
+            # A function to group the common logic of creating embeddings
+            def embed_create(input, vocabulary):
+                embedding_dims = int(math.sqrt(len(vocabulary)))
+                embedding_encoder = layers.Embedding(
+                    input_dim=len(vocabulary), output_dim=embedding_dims, name=input
+                )
+                self.embeddings[input] = embedding_encoder
+
+            if input == "target_movie_id" or input == "sequence_movie_ids":
+                # movie_id as the vocabulary for target_move_id and sequence_movie_ids
+                vocabulary = CATEGORICAL_FEATURES_WITH_VOCABULARY["movie_id"]
+                embed_create(input, vocabulary)
+            if input in CATEGORICAL_FEATURES_WITH_VOCABULARY:
+                # movie_id is only used to determine vocabulary size
+                if input == "movie_id":
+                    pass
+                else:
+                    # This includes user_id
+                    vocabulary = CATEGORICAL_FEATURES_WITH_VOCABULARY[input]
+                    embed_create(input, vocabulary)
+
+        # Dictionary of embeddings for features/inputs to the model
+        self.embeddings = dict()
+        for input in self.inputs:
+            embedding_helper(input)
+
+    def call(self, inputs):
+
+        # Returned concatenated list of features
+        encoded_transformer_features = []
+        encoded_other_features = []
+
+        # Helper function to create embeddings
+        def embedding_helper(input):
+            max_index = self.embeddings[input].input_dim - 1  # Clamp the indices
+            # torch had some index errors during embedding hence the clip function
+            encoded_feature = self.embeddings[input](
+                keras.ops.clip(inputs[input], 0, max_index)
+            )
+            return encoded_feature
+
+        ## Encode user features
+        for input in inputs:
+            if (input in CATEGORICAL_FEATURES_WITH_VOCABULARY):
+                if input == "user_id":
+                    if self.include_user_id:
+                        embedded_user_id = embedding_helper(input)
+                        encoded_other_features.append(embedded_user_id)
+                        continue
+
+                if input in USER_FEATURES:
+                    if self.include_user_features:
+                        embedded_user_feature = embedding_helper(input)
+                        encoded_other_features.append(embedded_user_feature)
+                        continue
+
+            # This function merely groups similar logic for include_movie_features=True,
+            # or given as include_movie_features=False
+            def movie_sequence_helper(encoded_sequence_movies):
+                # Create positional embedding.
+                positions = keras.ops.arange(start=0, stop=sequence_length - 1, step=1)
```

Comment:
Instead of using `keras.ops.*` everywhere, just do `from keras import ops` at the start -- it will shorten many lines